dm/master/scheduler/scheduler.go

```diff
@@ -239,6 +246,13 @@ func (s *Scheduler) Close() {
 	s.logger.Info("the scheduler has closed")
 }
 
+// CloseAllWorkers closes all the scheduler's workers.
+func (s *Scheduler) CloseAllWorkers() {
+	for _, worker := range s.workers {
```

Comment:
I think we should remove the worker from `s.workers`. Because in `(*Scheduler).AddWorker`, we skip adding worker if key exists (or reset `s.workers` when `(*Scheduler).Start`)

Reply:
https://github.com/pingcap/dm/blob/master/dm/master/scheduler/scheduler.go#L1732